File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode list1, int a, int b, ListNode list2) {
        ListNode parent = null;
        ListNode current = list1;
        
        if(current == null) {
            return list2;
        }
        else {
            
            for(int i = 0 ; i < a; i++) {
                parent = current;
                current = current.next;
            }
            
            parent.next = list2;
            while(parent.next != null)
            {   parent = parent.next;}
            
            for(int i = a; i < b ; i++) {
                current = current.next;
            }
            
            parent.next = current.next;
            
            return list1;
        }
    }
}

File: 1669-merge-in-between-linked-lists/1669-merge-in-between-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeInBetween(self, list1: ListNode, a: int, b: int, list2: ListNode) -> ListNode:
        parent = None
        current = list1
        
        if current == None:
            return list2
        else:
            for i in range(0 , a):
                parent = current
                current = current.next
            
            parent.next = list2
            
            while parent.next != None:
                parent = parent.next
                
            for i in range(a , b):
                current = current.next
                
            parent.next = current.next
            
            return list1
                